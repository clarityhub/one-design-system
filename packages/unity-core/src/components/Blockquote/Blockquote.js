import styled from '@emotion/styled';
import colors from '../../theme/colors';

const Blockquote = styled.blockquote`
    border-left: 4px solid ${colors.dark};
    padding-left: calc(2rem - 4px);
    font-style: italic;
    margin-left: -2rem;

    p + & {
        margin-top: 1.6rem;
    }

    & + p {
        margin-top: 1.6rem;
    }
`;

export default Blockquote;
