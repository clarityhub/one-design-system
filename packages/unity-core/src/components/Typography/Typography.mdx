---
name: Typography
menu: Theme
---

import { Playground, PropsTable } from 'docz';
import Typography from './Typography';

# Typography

## \<Typography /\> component

```js
import Typography from '@clarityhub/unity-core/components/Typography';
```

## Prop Types

<PropsTable of={Typography} />

## Examples

### Text versus Paragraph

By default, if no `type` is provided to `Typography`, then `text` is used.

If you want to have multiple paragraphs of readable text, use `type="p"`.

<Playground>
    <Typography>Lorem ipsum dolor sit amet, <i>consectetur adipiscing elit</i>. Donec odio ipsum, <strong>sodales et ligula sed</strong>, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
    <Typography type="p">Lorem ipsum dolor sit amet, <i>consectetur adipiscing elit</i>. Donec odio ipsum, <strong>sodales et ligula sed</strong>, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
</Playground>

### Headings

<Playground>
    <Typography type="h1">H1. The quick brown fox</Typography>
    <Typography type="h2">H2. The quick brown fox</Typography>
    <Typography type="h3">H3. The quick brown fox</Typography>
    <Typography type="h4">H4. The quick brown fox</Typography>
    <Typography type="h5">H5. The quick brown fox</Typography>
</Playground>

Without padding:

<Playground>
    <Typography type="h1" noPadding>H1. The quick brown fox</Typography>
    <Typography type="h2" noPadding>H2. The quick brown fox</Typography>
    <Typography type="h3" noPadding>H3. The quick brown fox</Typography>
    <Typography type="h4" noPadding>H4. The quick brown fox</Typography>
    <Typography type="h5" noPadding>H5. The quick brown fox</Typography>
</Playground>

### Text

<Playground>
    <Typography type="p">Lorem ipsum dolor sit amet, <i>consectetur adipiscing elit</i>. Donec odio ipsum, <em>sodales et ligula sed</em>, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
    <Typography type="p">Vestibulum quis facilisis dolor, <strong>a faucibus urna</strong>. Vestibulum lobortis massa in <b>lacus semper</b>, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>
    <Typography type="subtext">This is a set of subtext</Typography>
    <Typography type="sectionLabel">This is a section label</Typography>
</Playground>

### Kitchen Sink

<Playground>
    <Typography type="h1">H1. The quick brown fox</Typography>
    <Typography type="p">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec odio ipsum, sodales et ligula sed, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
    <Typography type="p">Vestibulum quis facilisis dolor, a faucibus urna. Vestibulum lobortis massa in lacus semper, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>

    <Typography type="h2">H2. The quick brown fox</Typography>

    <Typography type="p">Vestibulum quis facilisis dolor, a faucibus urna. Vestibulum lobortis massa in lacus semper, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>

    <Typography type="h2">H2. The quick brown fox</Typography>

    <Typography type="p">Vestibulum quis facilisis dolor, a faucibus urna. Vestibulum lobortis massa in lacus semper, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>
</Playground>