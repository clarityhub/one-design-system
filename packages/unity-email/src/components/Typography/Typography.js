import React from 'react';
import {
	MjmlText,
} from 'mjml-react';
import colors from '@clarityhub/unity-core/dist/theme/colors';

import { types, variants } from '../../theme/fonts';

const Typography = ({
	type = 'text',
	variant,
	color = 'dark',
	children,
	component: suggestedComponent = 'p',
	noMargin = false,
	noPadding = false,
	inline = false,
	css: userCss,
	center = false,
	...rest
}) => {
	const props = Object.assign(
		{},
		Boolean(type) ? types[type] : {},
		Boolean(variant) ? variants[variant] : {},
		Boolean(color) ? { color: colors[color] } : {},
		center ? { align: 'center' } : { align: 'left' },
	);

	return <MjmlText {...rest} {...props}>{children}</MjmlText>;
};

// XXX prop types

export default Typography;
