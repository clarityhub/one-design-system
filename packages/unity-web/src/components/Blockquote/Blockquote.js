import styled from '@emotion/styled';
import colors from '@clarityhub/unity-core/lib/utilities/color';

const Blockquote = styled.blockquote`
    border-left: 4px solid ${colors.dark};
    padding-left: calc(2rem - 4px);
    font-style: italic;
    margin-left: -2rem;

    p + & {
        margin-top: 1.6rem;
    }

    & + p {
        margin-top: 1.6rem;
    }
`;

export default Blockquote;
