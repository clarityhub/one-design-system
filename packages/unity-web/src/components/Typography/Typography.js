/** @jsx jsx */

import { bool, node, oneOf, string } from 'prop-types';
import styled from '@emotion/styled';
import { jsx } from '@emotion/core';
import colors from '@clarityhub/unity-core/lib/colors';
import { variants } from '@clarityhub/unity-core/lib/typography';

import { types } from '../../theme/fonts';

const noPaddingCss = `
    padding: 0;
`;

const noMarginCss = `
    margin-bottom: 0;
    margin-top: 0;

    & + & {
        margin-bottom: 0;
        margin-top: 0;
    }
`;

const Typography = ({
	type = 'text',
	variant,
	color = 'dark',
	children,
	component: suggestedComponent = 'p',
	noMargin = false,
	noPadding = false,
	inline = false,
	css: userCss,
	center = false,
	...rest
}) => {
	let styles = [
		Boolean(type) ? types[type]: {},
		Boolean(variant) ? variants[variant]: {},
		Boolean(color) ? { color: colors[color].default }: {},
		Boolean(noPadding) ? noPaddingCss: {},
		Boolean(noMargin) ? noMarginCss: {},
		Boolean(userCss) ? userCss: {},
		Boolean(inline) ? {
			display: 'inline-block',
		} : {},
		Boolean(center) ? {
			textAlign: 'center',
		} : {},
	];

	if (typeof suggestedComponent === 'string') {
		let component = suggestedComponent;
		if (suggestedComponent === 'p' && ['h1', 'h2', 'h3', 'h4', 'h5'].indexOf(type) !== -1) {
			component = type;
		}

		const Comp = styled[component](styles);
		return <Comp {...rest}>{children}</Comp>;
	} else {
		const Comp = suggestedComponent;
		return <Comp {...rest} css={styles}>{children}</Comp>;
	}
};

Typography.propTypes = {
	center: bool,
	children: node,
	color: oneOf(Object.keys(colors)),
	component: node,
	css: string,
	inline: bool,
	noMargin: bool,
	noPadding: bool,
	type: oneOf(Object.keys(types)),
	variant: oneOf(Object.keys(variants)),
};

export default Typography;
