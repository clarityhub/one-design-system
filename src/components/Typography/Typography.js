/** @jsx jsx */

import { bool, node, oneOf, string } from 'prop-types';
import styled from '@emotion/styled';
import { jsx, css } from '@emotion/core';

import { types, variants, colorStyles } from '../../theme/fonts';

const noPaddingCss = `
    padding: 0;
`;

const noMarginCss = `
    margin-bottom: 0;
    margin-top: 0;

    & + & {
        margin-bottom: 0;
        margin-top: 0;
    }
`;

const Typography = ({
    type = 'text',
    variant,
    color = 'black',
    children,
    component: suggestedComponent = 'p',
    noMargin = false,
    noPadding = false,
    css: userCss,
    ...rest,
}) => {
    let style = `
        ${Boolean(type) ? types[type] : ''}
        ${Boolean(variant) ? variants[variant] : ''}
        ${Boolean(color) ? colorStyles[color] : ''}
        ${Boolean(noPadding) ? noPaddingCss : ''}
        ${Boolean(noMargin) ? noMarginCss : ''}
        ${Boolean(userCss) ? userCss : ''}
    `;

    if (typeof suggestedComponent === 'string') {
        let component = suggestedComponent;
        if (suggestedComponent === 'p' && ['h1', 'h2', 'h3', 'h4', 'h5'].indexOf(type) !== -1) {
            component = type;
        }

        const Comp = styled[component](css`${style}`);
        return <Comp {...rest}>{children}</Comp>
    } else {
        const Comp = suggestedComponent;
        return <Comp {...rest} css={css`${style}`}>{children}</Comp>
    }
};

Typography.propTypes = {
    type: oneOf(Object.keys(types)),
    variant: oneOf(Object.keys(variants)),
    color: oneOf(Object.keys(colorStyles)),
    children: node,
    component: node,
    noMargin: bool,
    noPadding: bool,
    css: string,
};

export default Typography;
