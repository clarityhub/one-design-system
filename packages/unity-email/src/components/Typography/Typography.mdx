---
menu: Components
name: Typography
---

import { Playground, PropsTable } from 'docz';
import Typography from './Typography';

# Email Typography

```js
import Typography from '@clarityhub/unity-email/components/Typography';
```

## Prop Types

<PropsTable of={Typography} />
