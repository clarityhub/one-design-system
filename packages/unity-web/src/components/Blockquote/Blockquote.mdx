---
name: Blockquote
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Blockquote from './Blockquote';
import Typography from '../Typography';

# Blockquote

## \<Blockquote /\> component

```js
import Blockquote from '@clarityhub/unity-web/lib/components/Blockquote';
```

## Prop Types

<PropsTable of={Blockquote} />

## Examples

<Playground>
    <Typography>
        Normal paragraph text. Notice the alignment of the blockquote is
        to the left of the paragraph.
    </Typography>
    <Blockquote>
        <Typography>This is a block quote with some text</Typography>
        <Typography>There can be multiple paragraphs</Typography>
    </Blockquote>
</Playground>
