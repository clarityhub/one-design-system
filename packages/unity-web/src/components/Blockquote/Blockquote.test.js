import React from 'react';
import renderer from 'react-test-renderer';

import Typography from '../Typography';
import Blockquote from './Blockquote';

describe('<Blockquote />', () => {
	it('renders text', () => {
		const node = (
			<Blockquote>
				<Typography>This is a block quote with some text</Typography>
				<Typography>There can be multiple paragraphs</Typography>
			</Blockquote>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});