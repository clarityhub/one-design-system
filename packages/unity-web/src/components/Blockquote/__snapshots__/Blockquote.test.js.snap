// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blockquote /> renders text 1`] = `
<blockquote
  className="css-6fauep-Blockquote e9ltruz0"
>
  <p
    className="css-b04jnf"
  >
    This is a block quote with some text
  </p>
  <p
    className="css-b04jnf"
  >
    There can be multiple paragraphs
  </p>
</blockquote>
`;
